restapi/src/main/java/io/stargate/web/docsapi/service/DocsSchemaChecker.java

```diff
@@ -0,0 +1,66 @@
+package io.stargate.web.docsapi.service;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import io.stargate.web.docsapi.dao.DocumentDB;
+import io.stargate.web.docsapi.exception.ErrorCode;
+import io.stargate.web.docsapi.exception.ErrorCodeRuntimeException;
+import java.util.Objects;
+
+/**
+ * A helper class for checking that particular keyspace/table combinations exist and are valid
+ * document API collections.
+ */
+public class DocsSchemaChecker {
+  public static class KeyspaceAndTable {
+    private String keyspace;
+    private String table;
+
+    public KeyspaceAndTable(String keyspace, String table) {
+      this.keyspace = keyspace;
+      this.table = table;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+      KeyspaceAndTable that = (KeyspaceAndTable) o;
+      return keyspace.equals(that.keyspace) && table.equals(that.table);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(keyspace, table);
+    }
+  }
+
+  private final Cache<KeyspaceAndTable, Boolean> validatedDocCollections =
+      Caffeine.newBuilder().maximumSize(10_000).build();
+
+  public void checkTableExists(String keyspace, String table, DocumentDB db) {
+    db.tableExists(keyspace, table);
+  }
+
+  public boolean isDocumentsTable(String keyspace, String table, DocumentDB db) {
+    KeyspaceAndTable keyspaceAndTable = new KeyspaceAndTable(keyspace, table);
+    Boolean valid = validatedDocCollections.getIfPresent(keyspaceAndTable);
+    if (valid == null) {
+      valid = db.isDocumentsTable(keyspace, table);
+      validatedDocCollections.put(keyspaceAndTable, valid);
+    }
+    return valid;
+  }
+
+  public void checkValidity(String keyspace, String table, DocumentDB db) {
```

Comment:
are these public methods above needed at all? Or better said, can they be private? I would expect that everybody from outside just hit the `checkValidity`.

Reply:
I removed these in the most recent checks - also I made the assumption in the caching that if the schema is a valid collection, then the table must have existed - so I decided to cache them both together (if the table exists and all is well, it caches until the schema changes again)